docs/docs/noir/standard_library/meta/typ.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Type
+---
+
+`std::meta::typ` contains methods on the built-in `Type` type used for representing
+a type in the source program.
+
+## Methods
+
+### as_array
+
+#include_code as_array noir_stdlib/src/meta/typ.nr rust
+
+If this type is an array, return a pair of (element type, size type).
+
+Example:
+
+```rust
+comptime {
+    let array_type = quote { [Field; 3] }.as_type();
+    let (field_type, three_type) = array_type.as_array().unwrap();
+
+    assert(field_type.is_field());
+    assert_eq(three_type.as_constant().unwrap(), 3);
+}
+```
+
+### as_constant
+
+#include_code as_constant noir_stdlib/src/meta/typ.nr rust
+
+If this type is a constant integer (such as the `3` in the array type `[Field; 3]`),
+return the numeric constant.
+
+### as_integer
+
+#include_code as_integer noir_stdlib/src/meta/typ.nr rust
+
+If this is an integer type, return a boolean which is `true`
+if the type is signed, as well as the number of bits of this integer type.
+
+### as_slice
+
+#include_code as_slice noir_stdlib/src/meta/typ.nr rust
+
+If this is a slice type, return the element type of the slice.
+
+### as_struct
+
+#include_code as_struct noir_stdlib/src/meta/typ.nr rust
+
+If this is a struct type, returns the struct in addition to
+any generic arguments on this type.
+
+### as_tuple
+
+#include_code as_tuple noir_stdlib/src/meta/typ.nr rust
+
+If this is a tuple type, returns each element type of the tuple.
+
+### get_trait_impl
+
+#include_code get_trait_impl noir_stdlib/src/meta/typ.nr rust
+
+Retrieves the trait implementation that implements the given
+trait constraint for this type. If the trait constraint is not
+found, `None` is returned. Note that since the concrete trait implementation
+for a trait constraint specified from a `where` clause is unknown,
+this function will return `None` in these cases. If you only want to know
+whether a type implements a true, use `implements` instead.
```

Comment:
Confused by "whether a type implements a true"

Reply:
should be "implements a trait"